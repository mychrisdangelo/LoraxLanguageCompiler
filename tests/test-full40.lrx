/* 
 * Author: Chris D'Angelo
 * Test for root function
 */

int main()
{
	tree <int>t(2);
	tree <int>s(2);
	string v;
	v = "Hello Kira";
	t = 1[2, 3[4, 5]];
	s = t;

	print("should print 1[2, 3[4, 5]] = ", parent(parent(t%1%1)), "\n");
	print("should print 1[2, 3[4, 5]] = ", root(t%1%1), "\n");
	print("should print 1[2, 3[4, 5]] = ", root(s%1%1), "\n");
	print("should print Hello Kira = ", root(v%0%0%0), "\n");
	print("should print 1[2, 3, 4] = ", root(1[2, 3, 4]));
}