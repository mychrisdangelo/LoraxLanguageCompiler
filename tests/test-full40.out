should print 1[2, 3[4, 5]] = 1[2[null,null],3[4[null,null],5[null,null]]]
should print 1[2, 3[4, 5]] = 1[2[null,null],3[4[null,null],5[null,null]]]
should print 1[2, 3[4, 5]] = 1[2[null,null],3[4[null,null],5[null,null]]]
should print Hello Kira = Hello Kira
should print 1[2, 3, 4] = 1[2[null,null,null],3[null,null,null],4[null,null,null]]